app/controllers/hyrax/batch_edits_controller.rb

```diff
@@ -35,11 +35,19 @@ def check_for_empty!
     end
 
     def destroy_collection
+      # This flag is especially useful to catch non empty admin sets.
+      success = true
       batch.each do |doc_id|
         obj = ActiveFedora::Base.find(doc_id, cast: true)
-        obj.destroy
+        success = false if obj.destroy == false
       end
-      flash[:notice] = "Batch delete complete"
+
+      admin_set_type_name = Hyrax::CollectionType.find_or_create_admin_set_type.title
+      flash[:notice] = if flash[:notice].nil? && success
+                         I18n.t('hyrax.batch_edit.delete_complete.success')
+                       else
+                         I18n.t('hyrax.batch_edit.delete_complete.type_error', admin_set_name: admin_set_type_name)
```

Comment:
This will be returned if `flash[:notice]` is set, regardless of the value of `success`. Is that your intention?